src/Dimensions.cpp

```diff
@@ -377,14 +377,14 @@ void RangeDimension::ticks(const std::vector<double> &ticks) {
 
 
 double RangeDimension::tickAt(const ndsize_t index) const {
-
     size_t idx = check::fits_in_size_t(index, "Tick index exceeds memory (size larger than current system supports)");
 
-    vector<double> ticks = this->ticks();
-    if (idx >= ticks.size()) {
+    try {
+        vector<double> ticks = this->ticks(idx, 1);
+        return ticks[0];
+    } catch (...) {
```

Comment:
Ekks, wildcard exception catching... what if something that is "not out of bounds" goes wrong in any backend, this would convert that exception too.
I am always very sceptical of exceptions for control-flow (C++ is not python-ask-for-forgivness). For the `ticks[0]` access a simple `size()` check would suffice and the backend's `ticks(start, count)` function, since it is public, should handle out-of-bounds itself too (and throw the correct exception).